### ISSUE TITLE
Exception when player join the server

### ISSUE THREAD

**Issue by tudubucket** [ComuGamersES/qubo-blocker#2]
My server is using Paper 1.16.5, this is the warning log when a player join the server
```
[05:17:44 INFO]: UUID of player Player is b544b55a-878f-30ea-b559-50fd109dc3c4
[05:17:44 WARN]: [io.netty.util.concurrent.DefaultPromise] An exception was thrown by com.comugamers.quboblocker.lib.packetevents.injector.handlers.PacketEventsEncoder$$Lambda$7625/0x00000008024ffbb0.operationComplete()
java.lang.IllegalStateException: PacketEvents failed to find a ByteToMessageDecoder indexed 0 by its type in the net.minecraft.server.v1_16_R3.PacketDecoder class!
        at com.comugamers.quboblocker.lib.packetevents.util.reflection.ReflectionObject.getField(ReflectionObject.java:297) ~[?:?]
        at com.comugamers.quboblocker.lib.packetevents.util.reflection.ReflectionObject.read(ReflectionObject.java:159) ~[?:?]
        at com.comugamers.quboblocker.lib.packetevents.util.reflection.ReflectionObject.readObject(ReflectionObject.java:148) ~[?:?]
        at com.comugamers.quboblocker.lib.packetevents.injector.SpigotChannelInjector.changeConnectionState(SpigotChannelInjector.java:219) ~[?:?]
        at com.comugamers.quboblocker.lib.packetevents.manager.InternalPacketListener.lambda$onPacketSend$0(InternalPacketListener.java:77) ~[?:?]
        at com.comugamers.quboblocker.lib.packetevents.injector.handlers.PacketEventsEncoder.lambda$write$0(PacketEventsEncoder.java:100) ~[?:?]
        at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:577) ~[patched_1.16.5.jar:git-Paper-794]
        at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:551) ~[patched_1.16.5.jar:git-Paper-794]
        at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:490) ~[patched_1.16.5.jar:git-Paper-794]
        at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:615) ~[patched_1.16.5.jar:git-Paper-794]
        at io.netty.util.concurrent.DefaultPromise.setSuccess0(DefaultPromise.java:604) ~[patched_1.16.5.jar:git-Paper-794]
        at io.netty.util.concurrent.DefaultPromise.trySuccess(DefaultPromise.java:104) ~[patched_1.16.5.jar:git-Paper-794]
        at io.netty.util.internal.PromiseNotificationUtil.trySuccess(PromiseNotificationUtil.java:48) ~[patched_1.16.5.jar:git-Paper-794]
        at io.netty.channel.ChannelOutboundBuffer.safeSuccess(ChannelOutboundBuffer.java:717) ~[patched_1.16.5.jar:git-Paper-794]
        at io.netty.channel.ChannelOutboundBuffer.remove(ChannelOutboundBuffer.java:272) ~[patched_1.16.5.jar:git-Paper-794]
        at io.netty.channel.ChannelOutboundBuffer.removeBytes(ChannelOutboundBuffer.java:352) ~[patched_1.16.5.jar:git-Paper-794]
        at io.netty.channel.epoll.AbstractEpollChannel.doWriteBytes(AbstractEpollChannel.java:363) ~[patched_1.16.5.jar:git-Paper-794]
        at io.netty.channel.epoll.AbstractEpollStreamChannel.writeBytes(AbstractEpollStreamChannel.java:260) ~[patched_1.16.5.jar:git-Paper-794]
        at io.netty.channel.epoll.AbstractEpollStreamChannel.doWriteSingle(AbstractEpollStreamChannel.java:471) ~[patched_1.16.5.jar:git-Paper-794]
        at io.netty.channel.epoll.AbstractEpollStreamChannel.doWrite(AbstractEpollStreamChannel.java:429) ~[patched_1.16.5.jar:git-Paper-794]
        at io.netty.channel.AbstractChannel$AbstractUnsafe.flush0(AbstractChannel.java:930) ~[patched_1.16.5.jar:git-Paper-794]
        at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.flush0(AbstractEpollChannel.java:519) ~[patched_1.16.5.jar:git-Paper-794]
        at io.netty.channel.AbstractChannel$AbstractUnsafe.flush(AbstractChannel.java:897) ~[patched_1.16.5.jar:git-Paper-794]
        at io.netty.channel.DefaultChannelPipeline$HeadContext.flush(DefaultChannelPipeline.java:1372) ~[patched_1.16.5.jar:git-Paper-794]
        at io.netty.channel.AbstractChannelHandlerContext.invokeFlush0(AbstractChannelHandlerContext.java:750) ~[patched_1.16.5.jar:git-Paper-794]
        at io.netty.channel.AbstractChannelHandlerContext.invokeFlush(AbstractChannelHandlerContext.java:742) ~[patched_1.16.5.jar:git-Paper-794]
        at io.netty.channel.AbstractChannelHandlerContext.flush(AbstractChannelHandlerContext.java:728) ~[patched_1.16.5.jar:git-Paper-794]
        at io.netty.handler.flush.FlushConsolidationHandler.flushNow(FlushConsolidationHandler.java:204) ~[patched_1.16.5.jar:git-Paper-794]
        at io.netty.handler.flush.FlushConsolidationHandler.flush(FlushConsolidationHandler.java:138) ~[patched_1.16.5.jar:git-Paper-794]
        at io.netty.channel.AbstractChannelHandlerContext.invokeFlush0(AbstractChannelHandlerContext.java:750) ~[patched_1.16.5.jar:git-Paper-794]
        at io.netty.channel.AbstractChannelHandlerContext.invokeFlush(AbstractChannelHandlerContext.java:742) ~[patched_1.16.5.jar:git-Paper-794]
        at io.netty.channel.AbstractChannelHandlerContext.flush(AbstractChannelHandlerContext.java:728) ~[patched_1.16.5.jar:git-Paper-794]
        at io.netty.channel.ChannelDuplexHandler.flush(ChannelDuplexHandler.java:127) ~[patched_1.16.5.jar:git-Paper-794]
        at io.netty.channel.AbstractChannelHandlerContext.invokeFlush0(AbstractChannelHandlerContext.java:750) ~[patched_1.16.5.jar:git-Paper-794]
        at io.netty.channel.AbstractChannelHandlerContext.invokeWriteAndFlush(AbstractChannelHandlerContext.java:765) ~[patched_1.16.5.jar:git-Paper-794]
        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:790) ~[patched_1.16.5.jar:git-Paper-794]
        at io.netty.channel.AbstractChannelHandlerContext.writeAndFlush(AbstractChannelHandlerContext.java:758) ~[patched_1.16.5.jar:git-Paper-794]
        at io.netty.channel.DefaultChannelPipeline.writeAndFlush(DefaultChannelPipeline.java:1020) ~[patched_1.16.5.jar:git-Paper-794]
        at io.netty.channel.AbstractChannel.writeAndFlush(AbstractChannel.java:299) ~[patched_1.16.5.jar:git-Paper-794]
        at com.comphenix.protocol.injector.netty.channel.NettyChannelProxy.writeAndFlush(NettyChannelProxy.java:227) ~[?:?]
        at com.comphenix.protocol.injector.netty.channel.NettyChannelProxy.writeAndFlush(NettyChannelProxy.java:233) ~[?:?]
        at net.minecraft.server.v1_16_R3.NetworkManager.lambda$b$6(NetworkManager.java:323) ~[patched_1.16.5.jar:git-Paper-794]
        at com.comphenix.protocol.injector.netty.channel.NettyEventLoopProxy.lambda$proxyRunnable$2(NettyEventLoopProxy.java:48) ~[?:?]
        at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164) ~[patched_1.16.5.jar:git-Paper-794]
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472) ~[patched_1.16.5.jar:git-Paper-794]
        at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:384) ~[patched_1.16.5.jar:git-Paper-794]
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989) ~[patched_1.16.5.jar:git-Paper-794]
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[patched_1.16.5.jar:git-Paper-794]
        at java.lang.Thread.run(Thread.java:831) [?:?]
[05:17:44 INFO]: Player[/*.*.*.*:52892] logged in with entity id 90845 at ([world]590.5415258709493, 59.0, -247.30000001192093)
[05:17:44 INFO]: [+] Player 
```
Any further information can be provided if it is required (i.e. Plugins) cause I don't know what to do

**Comment by Pabszito** [comment]
I'll check this as soon as I can - we're currently in the middle of doing changes in our infrastructure and I don't have much time to check this right now. But I think it might just be because of an outdated PacketEvents version.

**Comment by Pabszito** [comment]
I kinda forgot about this, sorry. Could you please try with [this build](https://cdn.discordapp.com/attachments/798015458136817686/1088286832216784966/qubo-blocker-2023.0322.0.jar) instead? It seems to work fine for me:
```
[02:18:17 INFO]: Plugins (2): ProtocolLib, QuboBlocker
[02:20:19 INFO]: Checking version, please wait...
[02:20:19 INFO]: This server is running Paper version git-Paper-794 (MC: 1.16.5) (Implementing API version 1.16.5-R0.1-SNAPSHOT)
You are running the latest version
[02:21:24 WARN]: [QuboBlocker] Connection closed during server list ping from /181.230.160.162:62831 - are backend servers on this node exposed?
```

**Comment by tudubucket** [comment]
With ProtocolLib 5.0.0, QuboBlocker v2023.0322.0, ViaVersion + ViaBackward v4.6.1 (it's 4.6.2 now but i am too lazy to update them) it seems to work fine now
`packetevents late injection detected` may cause by i'm reloading your plugin by PlugMan, normal plugin enable by the server has no problem
```yaml
[05:56:32 INFO]: [QuboBlocker] Enabling QuboBlocker v2023.0322.0
[05:56:32 INFO]: [packetevents] Late injection detected, we missed packets so some functionality may break!
[05:56:32 INFO]: [QuboBlocker] QuboBlocker is now enabled and looking for connections!
[05:56:32 INFO]: [PlugMan] QuboBlocker has been reloaded.
[05:56:42 INFO]: UUID of player tudubucket is d84487f0-fdd3-3cb8-9d08-4826c9047ed2
[05:56:42 INFO]: [DiscordSRV] Player tudubucket joined with silent joining permission, not sending a join message
[05:56:42 INFO]: tudubucket[/*.*.*.*:44078] logged in with entity id 5217 at ([world_void]0.5, 22.0, 0.5)
[05:56:46 INFO]: [AuthMe] tudubucket logged in
[05:57:52 INFO]: tudubucket lost connection: Disconnected
[05:58:20 WARN]: [QuboBlocker] Connection closed during server list ping from /*.*.*.*:64746 - are backend servers on this node exposed?
[05:58:38 WARN]: [QuboBlocker] Connection closed during server list ping from /*.*.*.*:31965 - are backend servers on this node exposed?
```
